### PR TITLE
DROID-418: Update FoodSafe integrated products to latest requirements

### PR DESCRIPTION
**Pull request by jhavatar**
https://linear.app/combustion/issue/DROID-418/update-foods-integrated-mode-parameters-in-ble-framework

### Description
Update FoodSafe Integrated products to latest requirements in spreadsheet https://docs.google.com/spreadsheets/d/1oDlP5sua6cjwPci3izsTixdTFH7BUwA1mXkIBCCoqUk/edit#gid=1179861674

### Tech Notes
- Rename existing products to the new products where it makes sense
- Need to keep all deprecated products and their order, but rename them with prefix "Deprecated"
- Add the new products that did not match existing

**Does my renaming need to be more aggressive?**

### CODE REVIEW

File: combustion-android-ble/src/main/java/inc/combustion/framework/service/FoodSafeData.kt [hunk 1]
```diff
@@ -250,22 +266,28 @@ sealed class FoodSafeData {
...
-                        10u -> TurkeyGround
-                        11u -> Lamb
-                        12u -> LambGround
-                        13u -> FishAndShellfish
-                        14u -> FishAndShellfishGround
+                        1u -> Meats
+                        2u -> MeatsGround
+                        3u -> DeprecatedChicken
+                        4u -> PoultryGround
+                        5u -> DepracatedPork
```
Comment (nick-sasquatch) [inline combustion-android-ble/src/main/java/inc/combustion/framework/service/FoodSafeData.kt:273]: Could you fix this typo please?

File: combustion-android-ble/src/main/java/inc/combustion/framework/service/FoodSafeData.kt [hunk 2]
```diff
@@ -226,22 +236,28 @@ sealed class FoodSafeData {
...
+                        DeprecatedHamGround -> 8u
+                        DeprecatedTurkey -> 9u
+                        DeprecatedTurkeyGround -> 10u
+                        DeprecatedLamb -> 11u
+                        DeprecatedLambGround -> 12u
+                        Seafood -> 13u
+                        SeafoodGround -> 14u
                         DairyMilkLessThan10PercentFat -> 15u
-                        Game -> 16u
+                        Other -> 16u
```
Comment (nick-sasquatch) [inline combustion-android-ble/src/main/java/inc/combustion/framework/service/FoodSafeData.kt:254]: I'm a little sad that we lost the context of `Game` coming from a guy who was cooking kangaroo meat. :)

**Review by nick-sasquatch — APPROVED** [review]
This looks great--one nit about a typo, but feel free to merge after that's fixed. Thanks for adding the unit tests!

File: combustion-android-ble/src/main/java/inc/combustion/framework/service/FoodSafeData.kt (hunk 2 above)
Comment (jhavatar) [reply combustion-android-ble/src/main/java/inc/combustion/framework/service/FoodSafeData.kt:254]: lol, kangaroo does not look very appetizing - I imagine rat taste

**Pull request merged**